client/ayon_hiero/api/plugin.py

```diff
@@ -760,40 +787,36 @@ def _populate_attributes(self):
 
+       # publisher ui attribute inputs or default values if gui was not used
+        def get(key):
+            """Shorthand access for code readability"""
+            return self.pre_create_data.get(key)
+
         # ui_inputs data or default values if gui was not used
-        self.rename = self.ui_inputs.get(
-            "clipRename", {}).get("value") or self.rename_default
-        self.clip_name = self.ui_inputs.get(
-            "clipName", {}).get("value") or self.clip_name_default
-        self.hierarchy = self.ui_inputs.get(
-            "hierarchy", {}).get("value") or self.hierarchy_default
-        self.hierarchy_data = self.ui_inputs.get(
-            "hierarchyData", {}).get("value") or \
-            self.track_item_default_data.copy()
-        self.count_from = self.ui_inputs.get(
-            "countFrom", {}).get("value") or self.count_from_default
-        self.count_steps = self.ui_inputs.get(
-            "countSteps", {}).get("value") or self.count_steps_default
-        self.base_product_name = self.ui_inputs.get(
-            "productName", {}).get("value") or self.base_product_name_default
-        self.product_type = self.ui_inputs.get(
-            "productType", {}).get("value") or self.product_type_default
-        self.vertical_sync = self.ui_inputs.get(
-            "vSyncOn", {}).get("value") or self.vertical_sync_default
-        self.driving_layer = self.ui_inputs.get(
-            "vSyncTrack", {}).get("value") or self.driving_layer_default
-        self.review_track = self.ui_inputs.get(
-            "reviewTrack", {}).get("value") or self.review_track_default
-        self.audio = self.ui_inputs.get(
-            "audio", {}).get("value") or False
+        self.rename = self.pre_create_data.get("clipRename", self.rename_default)
+        self.clip_name = get("clipName") or self.clip_name_default
+        self.hierarchy = get("hierarchy") or self.hierarchy_default
+        self.count_from = get("countFrom") or self.count_from_default
+        self.count_steps = get("countSteps") or self.count_steps_default
+        self.subset_name = get("subsetName") or self.subset_name_default
+        self.subset_family = get("subsetFamily") or self.subset_family_default
+        self.vertical_sync = get("vSyncOn") or self.vertical_sync_default
+        self.driving_layer = get("vSyncTrack") or self.driving_layer_default
+        self.review_track = get("reviewTrack") or self.review_track_default
+        self.audio = get("audio") or False
+
+        self.hierarchy_data = {
+            key: get(key) or self.track_item_default_data[key]
+            for key in ["folder", "episode", "sequence", "track", "shot"]
+        }
 
         # build product name from layer name
-        if self.base_product_name == "<track_name>":
-            self.base_product_name = self.track_name
+        if self.subset_name == "<track_name>":
+            self.subset_name = self.track_name
```

Comment:
Old code was more correct for AYON terminology. No 'subset', use 'product'

Reply:
I've renamed subset for product everywhere.